api/base/testing/apicaller.go

```diff
@@ -49,6 +49,34 @@ type CheckArgs struct {
 	VersionIsZero bool
 }
 
+func checkArgs(c *gc.C, args *CheckArgs, facade string, version int, id, method string, inArgs, outResults interface{}) {
+	if args != nil {
```

Comment:
I just moved existing code - didn't want to change the functionality. I agree that moving the if args != nil out of that function and putting it in the caller would make it more obvious that args checking is only done if you pass in expected args. It seems like the original intent was for args = nil to be OK.